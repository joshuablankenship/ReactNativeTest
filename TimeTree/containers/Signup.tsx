import React, {Component} from 'react';
import {Image, Text, TextInput, TouchableOpacity, View} from 'react-native';
import RBSheet from 'react-native-raw-bottom-sheet';
import {styles} from '../styles/SignupStyles';
import CreateProfile from './CreateProfile';
interface Props {
  navigation: any;
}
export default class Signup extends Component<Props> {
  createProfileSheet: RBSheet | null | undefined;
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      isPopupTrue: true,
      errorEmail: '',
      errorPassword: '',
      hasErrors: false,
    };
  }
  validate = text => {
    console.log(text);
    let reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
    if (reg.test(text) === false) {
      return false;
    } else {
      return true;
    }
  };
  _checkForErrors = () => {
    let hasErrorsEmail = this.validate(this.state.email) == false || this.state.email.length ==0 
    let hasErrorsPassword = this.state.password.length < 8|| this.state.password.length > 32;
    let emailError = '';
    let passwordError= '';
    if (hasErrorsEmail && hasErrorsPassword) {
      emailError = 'Please Enter a valid email address';
      passwordError= 'Please Enter a valid password';
    }else if(hasErrorsPassword && !hasErrorsEmail){
      emailError = '';
      passwordError= 'Please Enter a valid password';
    }else if(hasErrorsEmail && !hasErrorsPassword){
      emailError = 'Please Enter a valid email address';
      passwordError= '';
    }
    console.log(this.state.password + this.state.errorEmail)

    this.setState({errorEmail: emailError, errorPassword: passwordError});

    if(!hasErrorsEmail && !hasErrorsPassword) {
      this.createProfileSheet.open();

    }
  };
  _onPressButtonGoogle = async () => {
    this.props.navigation.navigate('Main');
  };
  _openSignUpSheet = () => {
    this._checkForErrors();
  };
  _closeSignUpSheet = () => {
    this.props.closeSignupSheet();
  };
  _openProfileSheet = () => {
    this._checkForErrors();
  };
  _closeProfileSheet = () => {
    this.createProfileSheet.close();
  };
  render() {
    return (
      <View>
        <View style={styles.sheetHeader}>
          <TouchableOpacity
            style={styles.text}
            onPress={this._closeSignUpSheet}>
            <Image
              style={styles.a_logo}
              source={require('../assets/images/x.png')}
            />
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.text}
            onPress={this._closeSignUpSheet}>
            <Text style={styles.text}> Later</Text>
          </TouchableOpacity>
        </View>
        <Text style={styles.title}> Sign up</Text>
        <View style={styles.buttonContainer}>
          <TextInput
            style={
              this.state.errorEmail.length > 0
                ? styles.textinputError
                : styles.textinput
            }
            placeholder="Email Address"
            autoCorrect={false}
            onChangeText={text => this.setState({email: text})}
            value={this.state.email}
          />
          <Text style={styles.error}>{this.state.errorEmail}</Text>
          <TextInput
            style={
              this.state.errorPassword.length > 0
                ? styles.textinputError
                : styles.textinput
            }
            secureTextEntry
            placeholder="8 - 32 character password"
            onChangeText={text => this.setState({password: text})}
            value={this.state.password}
          />
           <Text style={styles.error}>{this.state.errorPassword}</Text>
          
          <TouchableOpacity
            style={styles.buttonGreen}
            onPress={this._openProfileSheet}>
            <Text style={styles.buttonText}>Create Account</Text>
          </TouchableOpacity>
          <Text style={styles.textCenter}>or</Text>
          <TouchableOpacity style={styles.buttonFacebook}>
            <Image
              style={styles.f_logo}
              source={require('../assets/images/facebook_logo.png')}
            />
            <Text style={styles.buttonText}> Login with Facebook</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.buttonApple}
            // onPress={() => this.RBSheet.open()}
          >
            <Image
              style={styles.a_logo}
              source={require('../assets/images/apple_logo.png')}
            />
            <Text style={styles.buttonText}> Sign in with Apple</Text>
          </TouchableOpacity>
        </View>
        <View>
          <RBSheet
            ref={ref => {
              this.createProfileSheet = ref;
            }}
            height={800}
            openDuration={250}
            customStyles={{
              container: {
                borderTopLeftRadius: 10,
                borderTopRightRadius: 10,
              },
            }}>
            <CreateProfile
              navigation={this.props.navigation}
              openProfileSheet={this._openProfileSheet}
              closeProfileSheet={this._closeProfileSheet}
              closeSignupSheet={this._closeSignUpSheet}
              closeLoginSheet={this.props.closeLoginSheet}
            />
          </RBSheet>
        </View>
      </View>
    );
  }
}
