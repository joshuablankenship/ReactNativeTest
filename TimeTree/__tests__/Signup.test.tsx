import {shallow, ShallowWrapper} from 'enzyme';
import React from 'react';
import {TouchableOpacity} from 'react-native';
import Signup from '../containers/Signup';
const createTestProps = (props: any) => ({
  navigation: {
    navigate: jest.fn(),
  },
  closeLoginSheet: jest.fn(),
  ...props,
});
describe('Test Signup Screen', () => {
  let wrapper: ShallowWrapper;
  let props: any;
  wrapper = shallow(<Signup {...props} />);
  props = createTestProps({
    navigate: () => {},
    closeLoginSheet: () => {},
  });
  it('It should match initial snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('It should display five touchable opacity components', () => {
    expect(wrapper.find(TouchableOpacity).length).toEqual(5);
  });
});
