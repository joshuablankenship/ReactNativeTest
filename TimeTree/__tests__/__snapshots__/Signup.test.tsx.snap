// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Signup Screen It should match initial snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "color": "#2FCC87",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 10,
        "marginTop": 5,
        "padding": 12,
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={Object {}}
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/x.png",
          }
        }
        style={
          Object {
            "height": 15,
            "width": 15,
          }
        }
      />
    </ForwardRef>
    <ForwardRef
      onPress={[Function]}
      style={Object {}}
    >
      <Text
        style={Object {}}
      >
         Later
      </Text>
    </ForwardRef>
  </View>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 50,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
     Sign up
  </Text>
  <View
    style={
      Object {
        "marginLeft": 40,
        "marginRight": 40,
        "marginTop": 10,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Email Address"
      rejectResponderTermination={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#f8f8f8",
          "borderWidth": 1,
          "fontWeight": "bold",
          "height": 53,
          "justifyContent": "center",
          "paddingHorizontal": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "color": "red",
          "fontWeight": "bold",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="8 - 32 character password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#f8f8f8",
          "borderWidth": 1,
          "fontWeight": "bold",
          "height": 53,
          "justifyContent": "center",
          "paddingHorizontal": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "color": "red",
          "fontWeight": "bold",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    />
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2FCC87",
          "borderRadius": 5,
          "color": "white",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginBottom": 10,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        Create Account
      </Text>
    </ForwardRef>
    <Text
      style={
        Object {
          "color": "gray",
          "fontWeight": "bold",
          "marginBottom": 40,
          "marginTop": 30,
          "textAlign": "center",
        }
      }
    >
      or
    </Text>
    <ForwardRef
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3A5A97",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
          "padding": 12,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/facebook_logo.png",
          }
        }
        style={
          Object {
            "height": 25,
            "width": 25,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
         Login with Facebook
      </Text>
    </ForwardRef>
    <ForwardRef
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
          "padding": 12,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/apple_logo.png",
          }
        }
        style={
          Object {
            "height": 15,
            "width": 15,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
         Sign in with Apple
      </Text>
    </ForwardRef>
  </View>
  <View>
    <RBSheet
      animationType="none"
      closeDuration={200}
      closeOnDragDown={false}
      closeOnPressBack={true}
      closeOnPressMask={true}
      customStyles={
        Object {
          "container": Object {
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
          },
        }
      }
      dragFromTopOnly={false}
      height={800}
      keyboardAvoidingViewEnabled={true}
      minClosingHeight={0}
      onClose={null}
      onOpen={null}
      openDuration={250}
    >
      <CreateProfile
        closeProfileSheet={[Function]}
        closeSignupSheet={[Function]}
        openProfileSheet={[Function]}
      />
    </RBSheet>
  </View>
</View>
`;
